### ISSUE TITLE
JVM Crash: No virtual method getParameterCount()I 

### ISSUE THREAD

**Issue by jonasbark** [Kotlin/kotlinx-io#35]
Stacktrace:

```
2019-02-01 15:03:04.932 1952-2744/[REDACTED] E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: [REDACTED], PID: 1952
    java.lang.NoSuchMethodError: No virtual method getParameterCount()I in class Ljava/lang/reflect/Method; or its super classes (declaration of 'java.lang.reflect.Method' appears in /system/framework/core-oj.jar)
        at kotlinx.coroutines.io.internal.CoroutinesEventLoop$FutureReflectionImpl.<clinit>(EventLoopExperimental.kt:36)
        at kotlinx.coroutines.io.internal.EventLoopExperimentalKt.<clinit>(EventLoopExperimental.kt:48)
        at kotlinx.coroutines.io.internal.EventLoopExperimentalKt.detectEventLoop(EventLoopExperimental.kt:51)
        at kotlinx.coroutines.io.jvm.javaio.BlockingAdapter.parkingLoop(Blocking.kt:228)
        at kotlinx.coroutines.io.jvm.javaio.BlockingAdapter.submitAndAwait(Blocking.kt:215)
        at kotlinx.coroutines.io.jvm.javaio.BlockingAdapter.submitAndAwait(Blocking.kt:186)
        at kotlinx.coroutines.io.jvm.javaio.InputAdapter.read(Blocking.kt:55)
        at okio.Okio$2.read(Okio.java:140)
        at okio.RealBufferedSink.writeAll(RealBufferedSink.java:105)
        at io.ktor.client.engine.okhttp.StreamRequestBody.writeTo(StreamRequestBody.kt:13)
        at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:72)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:254)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:200)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:776)
```

Device used: Huawei Honor 6x, Android 7.0
Version used: `0.1.4` (dependency of ktor `1.1.2`)

We're trying to find out more information.

Edit: as promised, more information:
- also happens on Samsung Galaxy S6, Android 7.0
- doesn't happen on an Android 9 device
- the crash happens when doing the following ktor call, where the server does respond with no body at all (no JSON, no String, nothing):

```
client.post<String>("$endpoint/api/post") {
                
}
``` 
it also happens when using ```post<Unit>```

**Comment by jonasbark** [comment]
It wasn't the server response at all but the request:

Example:
```
val result = client.post<String>("http://httpbin.org/anything") {
                this.body = MultiPartContent.build {
                    add("header", header.length.toLong(), header, ContentType.Application.Json, filename = "blubb")
                    add("payload", payload.length.toLong(), payload, ContentType.Application.Json, filename = "blob")
                }
            }
```
Where the MultiPartContent is implemented like this: https://gist.github.com/jonasbark/3a0054dcf0c8ac9a896a0d805574914c

**Comment by cy6erGn0m** [comment]
Fixed in 0.1.5

**Comment by SergioIgreja** [comment]
I'm having this issue when sending a post request with ktor. Any ideas on how to solve it?
